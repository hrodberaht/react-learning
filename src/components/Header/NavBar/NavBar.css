ul {
    list-style-type: none;
    padding: 0;
    overflow: hidden;
    background-color: #333;
}

ul li {
    float: left;
}

ul li a {
    display: block;
    color: white;
    text-align: center;
    padding: 14px 16px;
    text-decoration: none;
}

ul li a:hover {
    background-color: #111;
}

a.active {
    color: red;
}