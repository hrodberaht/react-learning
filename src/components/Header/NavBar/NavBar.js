import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';

import ConnectedAuthorization from './Authorization/Authorization';

export default class NavBar extends Component {
  render() {
    return (
      <nav>
        <div className="navbar">
          <ul className="navbar__list">
            <li className="navbar__item">
              <NavLink className="navbar__item__link" to="/" exact>
                Home
              </NavLink>
            </li>
            <li className="navbar__item">
              <NavLink className="navbar__item__link" to="/cards">
                Cards
              </NavLink>
            </li>
            <li className="navbar__item">
              <NavLink className="navbar__item__link" to="/products">
                Products
              </NavLink>
            </li>
            <li className="navbar__item">
              <ConnectedAuthorization
                withAuth={logOut => (
                  <NavLink className="navbar__item__link" to="/login" onClick={logOut}>
                    Log out
                  </NavLink>
                )}
                withoutAuth={(
                  <NavLink className="navbar__item__link" to="/login">
                    Login
                  </NavLink>
)}
              />
            </li>
          </ul>
        </div>
        <div className="navbar--narrow">
          <i className="navbar--narrow__ico fa fa-bars fa-2x" />
          <div className="menu">
            <ul className="menu__list">
              <li className="menu__item">
                <NavLink className="menu__item__link" to="/" exact>
                  Home
                </NavLink>
              </li>
              <li className="menu__item">
                <NavLink className="menu__item__link" to="/cards">
                  Cards
                </NavLink>
              </li>
              <li className="menu__item">
                <NavLink className="menu__item__link" to="/products">
                  Products
                </NavLink>
              </li>
              <li className="menu__item">
                <ConnectedAuthorization
                  withAuth={logOut => (
                    <NavLink className="menu__item__link" to="/login" onClick={logOut}>
                      Log out
                    </NavLink>
                  )}
                  withoutAuth={(
                    <NavLink className="menu__item__link" to="/login">
                      Login
                    </NavLink>
)}
                />
              </li>
            </ul>
          </div>
        </div>
      </nav>
    );
  }
}
