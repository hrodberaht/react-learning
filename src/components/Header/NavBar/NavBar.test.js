import React from 'react';
import { shallow } from 'enzyme';

import { NavBar } from './NavBar';

describe('<NavBar />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<NavBar />);
  });
  it('should render component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
