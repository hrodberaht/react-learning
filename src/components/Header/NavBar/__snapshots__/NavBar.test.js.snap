// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> should render component 1`] = `
<nav>
  <div
    className="navbar"
  >
    <ul
      className="navbar__list"
    >
      <li
        className="navbar__item"
      >
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="navbar__item__link"
          exact={true}
          to="/"
        >
          Home
        </NavLink>
      </li>
      <li
        className="navbar__item"
      >
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="navbar__item__link"
          to="/cards"
        >
          Cards
        </NavLink>
      </li>
      <li
        className="navbar__item"
      >
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="navbar__item__link"
          to="/products"
        >
          Products
        </NavLink>
      </li>
      <li
        className="navbar__item"
      >
        <withRouter(Connect(Authorization))
          withAuth={[Function]}
          withoutAuth={
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="navbar__item__link"
              to="/login"
            >
              Login
            </NavLink>
          }
        />
      </li>
    </ul>
  </div>
  <div
    className="navbar--narrow"
  >
    <i
      className="navbar--narrow__ico fa fa-bars fa-2x"
    />
    <div
      className="menu"
    >
      <ul
        className="menu__list"
      >
        <li
          className="menu__item"
        >
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="menu__item__link"
            exact={true}
            to="/"
          >
            Home
          </NavLink>
        </li>
        <li
          className="menu__item"
        >
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="menu__item__link"
            to="/cards"
          >
            Cards
          </NavLink>
        </li>
        <li
          className="menu__item"
        >
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="menu__item__link"
            to="/products"
          >
            Products
          </NavLink>
        </li>
        <li
          className="menu__item"
        >
          <withRouter(Connect(Authorization))
            withAuth={[Function]}
            withoutAuth={
              <NavLink
                activeClassName="active"
                aria-current="page"
                className="menu__item__link"
                to="/login"
              >
                Login
              </NavLink>
            }
          />
        </li>
      </ul>
    </div>
  </div>
</nav>
`;
